src/connectionController.ts

```diff
@@ -161,7 +166,49 @@ export default class ConnectionController {
     });
   }
 
+  async openPresetConnectionsSettings(
+    originTreeItem: ConnectionTreeItem | undefined
+  ): Promise<void> {
+    this._telemetryService.trackPresetConnectionEdited({
+      source: DocumentSource.DOCUMENT_SOURCE_TREEVIEW,
+      source_details: originTreeItem ? 'tree_item' : 'header',
+    });
+    let source: ConnectionSource | undefined = originTreeItem?.source;
+    if (!source) {
+      const mdbConfiguration = vscode.workspace.getConfiguration('mdb');
+
+      const presetConnections = mdbConfiguration?.inspect(
+        'presetSavedConnections'
+      );
+
+      if (presetConnections?.workspaceValue) {
+        source = 'workspaceSettings';
+      } else if (presetConnections?.globalValue) {
+        source = 'globalSettings';
+      } else {
+        // If no preset settings exist in workspace and global scope,
+        // set a default one inside the workspace and open it.
+        source = 'workspaceSettings';
+        await mdbConfiguration.update('presetSavedConnections', [
+          {
+            name: 'Preset Database',
+            connectionString: 'mongodb://localhost:27017',
+          },
+        ]);
+      }
+    }
+    if (originTreeItem?.source === 'globalSettings') {
+      await vscode.commands.executeCommand('workbench.action.openSettingsJson');
+    } else {
+      await vscode.commands.executeCommand(
+        'workbench.action.openWorkspaceSettingsFile'
+      );
+    }
+  }
+
   async loadSavedConnections(): Promise<void> {
+    this._connections = Object.create(null);
```

Comment:
Is this fixing a bug where deleted saved connections would show up because they're not overridden in the for loop? Couldn't spot a test that verifies the behavior before was incorrect - am I just blind or should we add one?

Reply:
Yeah, this hasn't been an issue because we'd only run `loadSavedConnections` at startup / when `this._connections` would be not yet defined. Now with the reloading on save this becomes more obvious. I'll add a test.